### PR TITLE
Change pygithub commmit enttiy to pydriller commit

### PR DESCRIPTION
**Pull request by xtuchyna**
## Related Issues and Dependencies
To avoid requesting GitHub API for commit analysis, we can drill through the commits using `pydriller` :toolbox: !

## This introduces a breaking change

- [ ] Yes
- [X] No

## This Pull Request implements

… Explain your changes.

### CODE REVIEW

**Review by harshad16 — COMMENTED** [review]
/lgtm
nice work :+1:

File: srcopsmetrics/entities/commit.py [hunk 1]
```diff
@@ -42,29 +65,41 @@ class Commit(Entity):
...
-        return {f.filename: f.patch for f in commit.files}
+        patches = {}
+        for mod in commit.modified_files:
+            changed_methods = [method.name for method in mod.changed_methods]
+            patches[mod.filename] = {
+                "type": mod.change_type,
+                "changed_methods": changed_methods,
+                "patch_added": mod.diff_parsed["added"],
+                "patch_deleted": mod.diff_parsed["deleted"],
+            }
```
Comment (xtuchyna) [inline srcopsmetrics/entities/commit.py:90]: Patch information

File: srcopsmetrics/entities/commit.py [hunk 2]
```diff
@@ -42,29 +65,41 @@ class Commit(Entity):
...
+
+        self.stored_entities[commit.hash] = {
+            "pull_request": pull_request_ids,
+            "author": author_login,
+            "message": commit.msg,
+            "date": commit.committer_date.timestamp(),
+            "additions": commit.insertions,
+            "deletions": commit.deletions,
+            "files": commit.files,
+            # "patches": patches,
```
Comment (pacospace) [inline srcopsmetrics/entities/commit.py:100]: why is it commented?

**Review by pacospace — COMMENTED** [review]
One comment, otherwise lgtm!

File: srcopsmetrics/entities/commit.py (hunk 2 above)
Comment (xtuchyna) [reply srcopsmetrics/entities/commit.py]: nice catch :)

**Review by pacospace — APPROVED** [review]
nice improvement!! 💯 lgtm!

**Pull request merged**